src/modules/checkboxes.js

```diff
@@ -0,0 +1,45 @@
+// checkboxes.js
+import { saveTasksToLocalStorage } from './taskFunctions.js';
+
+// Add checkbox functionality to tasks
+export function addCheckboxFunctionality(taskList) {
+  taskList.addEventListener('change', (event) => {
+    if (event.target.classList.contains('task-checkbox')) {
+      const listItem = findParentListItem(event.target);
+      if (listItem) {
+        const taskText = listItem.querySelector('.task-item span'); // Updated selector
+        taskText.classList.toggle('completed', event.target.checked);
+
+        saveTasksToLocalStorage(taskList);
+      }
+    }
+  });
+}
+
+export function addClearCompletedFunctionality(taskList) {
+  const clearButton = document.getElementById('delete-selected');
+
+  clearButton.addEventListener('click', () => {
+    const completedTasks = Array.from(taskList.querySelectorAll('.completed'));
+
+    completedTasks.forEach((task) => {
+      const listItem = findParentListItem(task);
+      if (listItem) {
+        listItem.remove();
+      }
+    });
+
+    saveTasksToLocalStorage(taskList);
+  });
+}
```

Comment:
- [ ] Great work implementing the clear all completed function. However, based on the requirements, you ought to use the JavaScript `filter()` method to remove all completed tasks. Kindly use this method. 👍 😎